### PR TITLE
refactor(allthethings): refactor pipeline, fix linting, etc.

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/SURF-673

* moved nunjucks libs to `docs/_nunjucks`
  * they only pertain to documentation
* cleaned up/refactored `_config.js` to place relevant configs directly
  into the scripts that need them
* reorganized assets in `dist/` to match most open source projects
  * `dist/scripts/*`
  * `dist/styles/*`
* removed sourcemap generation
  * It is impossible to generate correct sourceMapURL that will work for
    every application.
  * The only way to make it work is with inline sourcemaps, but the
    amount of content this strategy adds counteracts the benefits of
    minification (results in file sizes larger than unminified).
* documentation only uses unminified assets
* boilerplate layouts use minified assets (when possible)
* modified the generated `docs.css` header
* fixed active nav link bug with `<hx-menu>` (tl;dr - trailing space in data)

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

File: _config.js [hunk 1]
```diff
@@ -9,8 +9,8 @@ CONFIG.root = path.resolve(__dirname);// absolute path to project directory
 CONFIG.publicDir = 'public';
 CONFIG.sourceDir = 'src';
 CONFIG.docsDir = 'docs';
-CONFIG.templateDir = `${CONFIG.docsDir}/_templates`;
```
Comment (CITguy) [inline _config.js]: Moved to `lib/nunjucks.js`

File: _config.js [hunk 2]
```diff
@@ -30,43 +30,6 @@ CONFIG.less = {
         `${CONFIG.docsDir}/styles`,
         `${CONFIG.sourceDir}/helix-ui/styles`
     ],
-    // [src, dest]
-    files: [
-        [ `${CONFIG.sourceDir}/helix-ui.less`, 'dist/helix-ui.css' ],
-        [ `${CONFIG.docsDir}/docs.less`, 'public/docs.css' ],
-    ],
```
Comment (CITguy) [inline _config.js]: Split amongst `lib/compile.js` and `lib/generate.js`

File: _config.js [hunk 3]
```diff
@@ -30,43 +30,6 @@ CONFIG.less = {
...
-
-// TODO: move to lib/webpack.js
-// Configuration for the webpack build tool
-CONFIG.webpack = {
-    entry: `${CONFIG.root}/${CONFIG.docsDir}/docs.js`,
-    output: {
-        path: path.resolve(__dirname, 'public'),
-        filename: 'docs.js',
-    }
-};
```
Comment (CITguy) [inline _config.js]: config moved to `lib/generate.js`

File: _config.js [hunk 4]
```diff
@@ -30,43 +30,6 @@ CONFIG.less = {
...
-
-// Configuration for generating static documentation
-CONFIG.docs = {
-    files: [
-        '*.html',
-        '**/*.html',
-    ],
```
Comment (CITguy) [inline _config.js]: `files` config moved to `lib/generate.js`

File: _config.js [hunk 5]
```diff
@@ -30,43 +30,6 @@ CONFIG.less = {
...
-    // settings for deployment to github pages
-    ghPages: {
-        type: 'git',
-        repo: 'git@github.com:rackerlabs/helix-ui.git',
-        branch: 'gh-pages',
-        message: `Docs Updated: {{ now('YYYY-MM-DD HH:mm:ss') }}`,
-        url: 'https://rackerlabs.github.io/helix-ui'
-    }
```
Comment (CITguy) [inline _config.js]: `ghPages` removed. Unused.

File: _config.js [hunk 6]
```diff
@@ -30,43 +30,6 @@ CONFIG.less = {
...
-};
-
-// Files and directories to copy to publicDir
-// All paths are relative to {CONFIG.sourceDir}
-CONFIG.copy = {
-    dirs: [ 'images' ]
```
Comment (CITguy) [inline _config.js]: moved to `lib/copy.js`

File: bin/build.js [hunk 7]
```diff
@@ -1,30 +0,0 @@
-#!/usr/bin/env node
```
Comment (CITguy) [inline bin/build.js]: Replaced with `lib/generate.js`

File: bin/clean.js [hunk 8]
```diff
@@ -1,15 +0,0 @@
-#!/usr/bin/env node
```
Comment (CITguy) [inline bin/clean.js]: Replaced with `bin/cleanPublic.js`

File: bin/copy.js [hunk 9]
```diff
@@ -1,28 +0,0 @@
-#!/usr/bin/env node
```
Comment (CITguy) [inline bin/copy.js]: Converted to `lib/copy.js`

File: bin/publish.js [hunk 10]
```diff
@@ -1,36 +0,0 @@
-#!/usr/bin/env node
```
Comment (CITguy) [inline bin/publish.js]: Replaced with `bin/ghpages.js` and `lib/publish.js`

File: bin/start.js [hunk 11]
```diff
@@ -1,23 +1,14 @@
...
+const exec = require('child_process').exec;
+const { compileScripts, compileStyles} = require('../lib/compile');
+const { copyDist } = require('../lib/copy');
+const { generateAll } = require('../lib/generate');
 
-// Clean out old assets
-Clean.cleanSync();
-
-// Build new assets
-Build.buildSync();
```
Comment (CITguy) [inline bin/start.js]: Handled in `prestart` task.

File: docs/_data/nav.json5 [hunk 12]
```diff
@@ -61,7 +61,7 @@
             { label: '<hx-checkbox>', path: 'hx-checkbox' },
             { label: '<hx-disclosure>', path: 'hx-disclosure' },
             { label: '<hx-icon>', path: 'hx-icon' },
-            { label: '<hx-menu>', path: 'hx-menu ' },
```
Comment (CITguy) [inline docs/_data/nav.json5]: fixes an "active" nav link issue (go figure 😄 )

File: package.json [hunk 13]
```diff
@@ -50,17 +53,22 @@
...
+    "preghpages": "yarn generate",
+    "ghpages": "bin/ghpages.js",
+    "lint": "eslint -c .eslintrc.json source #FIXME",
     "postinstall": "cd test; yarn install; yarn run webdriver:update; cd -;",
-    "publish": "bin/publish.js",
+    "postpublish": "yarn ghpages",
+    "postversion": "yarn compile",
+    "prepublishOnly": "yarn generate",
     "rollup": "rollup -c rollup.config.js",
-    "rollup:watch": "rollup -c rollup.config.js --watch",
```
Comment (CITguy) [inline package.json]: Removed. Similar functionality handled by the `start` task.

File: README.md [hunk 14]
```diff
@@ -6,25 +6,3 @@
 0. [install test dependencies](https://github.com/rackerlabs/helix-ui/wiki/Testing#system-requirements)
 0. `yarn install` to install project dependencies
 0. `yarn start` to begin working
-
-
-### Key Files/Directories
```
Comment (CITguy) [inline README.md]: Removed. This section hasn't been correct for a while. If we need to document the yarn tasks, we'll do so in the project wiki.

File: bin/start.js [hunk 15]
```diff
@@ -1,23 +1,14 @@
...
-const _ = require('lodash');
-
-const Build = require('./build');
-const Copy = require('./copy');
 const CONFIG = require('../_config');
-const Clean = require('./clean');
+const _ = require('lodash');
 const browserSync = require('browser-sync').create();
+const exec = require('child_process').exec;
+const { compileScripts, compileStyles} = require('../lib/compile');
```
Comment (nimbinatus) [inline bin/start.js]: Add space before `}`

File: docs/components/layouts/horizontal-layout-template.html [hunk 16]
```diff
@@ -100,14 +100,14 @@ <h2 class="hxSectionTitle">Right Panel</h2>
...
-  <!-- HelixUI Behavior -->
-  <script src="dist/helix-ui.browser.min.js"></script>
+  <!-- FIXME: point to HelixUI behavior in node_modules/helix-ui -->
+  <script src="dist/scripts/helix-ui.browser.min.js"></script>
 
   <a id="bottom"></a>
 
-  <!-- CHANGE ME: point to your application logic -->
-  <script src="docs.js"></script>
+  <!-- TODO: Add your application logic here! -->
```
Comment (nimbinatus) [inline docs/components/layouts/horizontal-layout-template.html]: Change to `FIXME`

File: docs/components/layouts/vertical-layout-template.html [hunk 17]
```diff
@@ -95,14 +95,14 @@ <h1 class="hxPageTitle">Page Content</h1>
...
-  <!-- HelixUI Behavior -->
-  <script src="dist/helix-ui.browser.min.js"></script>
+  <!-- FIXME: point to HelixUI behavior in node_modules/helix-ui -->
+  <script src="dist/scripts/helix-ui.browser.min.js"></script>
 
   <a id="bottom"></a>
 
-  <!-- CHANGE ME: point to your application logic -->
-  <script src="docs.js"></script>
+  <!-- TODO: Add your application logic here! -->
```
Comment (nimbinatus) [inline docs/components/layouts/vertical-layout-template.html]: Change to `FIXME`

File: .travis.yml [hunk 18]
```diff
@@ -16,8 +16,7 @@ before_script:
 - export DISPLAY=:99.0
 - sh -e /etc/init.d/xvfb start
 - jdk_switcher use oraclejdk8
-- yarn rollup
 - yarn start > /dev/null 2>&1 &
-- sleep 5
+- sleep 10
```
Comment (CITguy) [inline .travis.yml]: `yarn start` takes a little longer because we added rollup execution to the process

**Pull request merged**